### ISSUE TITLE
mail unable to send, Error 500

### ISSUE THREAD

**Issue by expert-geeks** [nextcloud/mail#2106]
### Expected behavior
Mail client is able to reply and send email

### Actual behavior
Email account and emails are 'viewable'. Icons/links to 'reply' are missing when viewing received mail. Attempting to send (by clicking 'New Mail') results in 

"Error sending your message
Error: Request failed with status code 500
Go back | Retry"

Incidentally, contacts from the contacts app are not offered when creating the email.

### Mail app

**Mail app version:** 0.17.0

**Mailserver or service:** Postfix/Dovecot

### Server configuration

**Operating system**: Debian 10

**Web server:** Nginx

**Database:** MariaDB Version: 10.4.8

**PHP version:** Version: 7.3.10

**Nextcloud Version:** 17.0.0

#### Client configuration
**Browser:** Firefox 69.0.3/Chrome Version 78.0.3904.50 (Official Build) beta (64-bit)

**Operating system:** Windows

**Nextcloud Log:**
[mail] Error: Horde_Mime_Exception: mail() returned failure. at <<closure>>

 0. /var/www/nextcloud/apps/mail/vendor/pear-pear.horde.org/Horde_Mime/Horde/Mime/Mail.php line 479
    Horde_Mime_Part->send("info@user.co.uk", Horde_Mime_Headers {}, Horde_Mail_Transport_Mail {sep: "\n"})
 1. /var/www/nextcloud/apps/mail/lib/Account.php line 214
    Horde_Mime_Mail->send(Horde_Mail_Transport_Mail {sep: "\n"}, false, false)
 2. /var/www/nextcloud/apps/mail/lib/Service/MailTransmission.php line 117
    OCA\Mail\Account->sendMessage(OCA\Mail\Model\Message {}, Horde_Mail_Transport_Mail {sep: "\n"}, null)
 3. /var/www/nextcloud/apps/mail/lib/Controller/AccountsController.php line 282
    OCA\Mail\Service\MailTransmission->sendMessage("mother", OCA\Mail\Model\NewMessageData {}, OCA\Mail\Model\RepliedMessageData {}, null, null)
 4. /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 170
    OCA\Mail\Controller\AccountsController->send(1, "FWD: report sum ... 0", "\n\n\"www.block ... e", "info@user.co.uk", "", "", null, null, null, [], null)
 5. /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 99
    OC\AppFramework\Http\Dispatcher->executeController(OCA\Mail\Controller\AccountsController {}, "send")
 6. /var/www/nextcloud/lib/private/AppFramework/App.php line 126
    OC\AppFramework\Http\Dispatcher->dispatch(OCA\Mail\Controller\AccountsController {}, "send")
 7. /var/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php line 47
    OC\AppFramework\App::main("OCA\\Mail\\Cont ... r", "send", OC\AppFramework\ ... {}, {accountId: "1", ... "})
 8. <<closure>>
    OC\AppFramework\Routing\RouteActionHandler->__invoke({accountId: "1", ... "})
 9. /var/www/nextcloud/lib/private/Route/Router.php line 297
    undefinedundefinedcall_user_func(OC\AppFramework\ ... {}, {accountId: "1", ... "})
10. /var/www/nextcloud/lib/base.php line 1000
    OC\Route\Router->match("/apps/mail/api/accounts/1/send")
11. /var/www/nextcloud/index.php line 42
    OC::handleRequest()

POST /apps/mail/api/accounts/1/send
from x.x.x.x by user at 2019-10-13T11:23:21+01:00

[mail] Fatal: Sending mail failed: mail() returned failure.

**Comment by expert-geeks** [comment]
Incidentally, email is always 'new'. it doesn't get marked as 'read' after reading in the app

**Comment by ChristophWurst** [comment]
> mail() returned failure

The Nextcloud mail app offers an extensive logging system to make it easier identifying and tracking down bugs. Please enable [debug mode](https://docs.nextcloud.com/server/13/developer_manual/general/devenv.html#enabling-debug-mode) and set the log [level to debug](https://docs.nextcloud.com/server/stable/admin_manual/configuration_server/logging_configuration.html) in your admin settings. Then, try to reproduce your issue and take another look at `data/nextcloud.log`, `data/horde_imap.log` and `data/horde_smtp.log`. Please remove all sensitive information.

**Comment by ChristophWurst** [comment]
Is your webserver/php allowed to send mail?

**Comment by EmiSchurr** [comment]
I've encountered this issue as well, and it turns out to be a bug with the way an e-mail address is entered in the recipiant line. It doesn't "save" anything you've put in to that line, and when trying to send throwing an error because there is no recipiant address.
A quick an dirty way of getting around this is to copy the e-mail address to your clipboard, then reloading the page and quickly pasting the address into the recipiant line, so that the whole address is shown again and ready to be selected as usual beneath the recipiant line.

**Comment by ChristophWurst** [comment]
See #1973

**Comment by bcutter** [comment]
5 minutes after giving the mail app a shot I ended up in the same situation with the same error message. I still can not send an eMail. The workaround of EmiSchurr is also not working. What the...? A mail app not being able to send mails, great. Any help appreciated.

**Comment by ChristophWurst** [comment]
@bcutter mind giving you details about your setup? From the sparse description it could be virtually anything.

The Nextcloud mail app offers an extensive logging system to make it easier identifying and tracking down bugs. Please enable [debug mode](https://docs.nextcloud.com/server/latest/developer_manual/general/devenv.html#enabling-debug-mode) and set the log [level to debug](https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/logging_configuration.html) in your admin settings. Then, try to reproduce your issue and take another look at `data/nextcloud.log` and `data/horde_smtp.log`.

**Comment by Supergamerx3000** [comment]
I discovered the same error and you cannot send an email if you only enter the recipient in the bcc. Error 500 appears. This works in Outlook and other e-mail programs.

[mail] Error: OCA\Mail\Exception\ServiceException: Could not send message: Server is not accepting SMTP connections. at <<closure>>

0. /var/www/html/nextcloud/apps/mail/lib/Controller/AccountsController.php line 345
   OCA\Mail\Service\MailTransmission->sendMessage(OCA\Mail\Model\NewMessageData {}, null, null, 1762)
1. /var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 170
   OCA\Mail\Controller\AccountsController->send(7, "Test 10:27", "", "", "", "empfaenger@domain.de", false, 1762, null, null, [], null)
2. /var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 99
   OC\AppFramework\Http\Dispatcher->executeController(OCA\Mail\Controller\AccountsController {}, "send")
3. /var/www/html/nextcloud/lib/private/AppFramework/App.php line 125
   OC\AppFramework\Http\Dispatcher->dispatch(OCA\Mail\Controller\AccountsController {}, "send")
4. /var/www/html/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php line 47
   OC\AppFramework\App::main("OCA\\Mail\\Cont ... r", "send", OC\AppFramework\ ... {}, {accountId: "7", ... "})
5. <<closure>>
   OC\AppFramework\Routing\RouteActionHandler->__invoke({accountId: "7", ... "})
6. /var/www/html/nextcloud/lib/private/Route/Router.php line 299
   call_user_func(OC\AppFramework\ ... {}, {accountId: "7", ... "})
7. /var/www/html/nextcloud/lib/base.php line 1013
   OC\Route\Router->match("/apps/mail/api/accounts/7/send")
8. /var/www/html/nextcloud/index.php line 38
   OC::handleRequest()

POST /apps/mail/api/accounts/7/send
from 771.256.291.559 by administrator-abc at 2020-09-17T08:27:52+00:00

**Comment by ChristophWurst** [comment]
> Server is not accepting SMTP connections.

Open a separate ticket. You have a different problem.

**Comment by agboom** [comment]
I'm having similar symptoms with sending mail. I've tried collecting some logs with the help of [General troubleshooting](https://docs.nextcloud.com/server/latest/admin_manual/issues/general_troubleshooting.html#nextcloud-logfiles) and [Logging](https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/logging_configuration.html), but I'm not getting very much. Maybe I'm missing something?

I've added these settings to `config.php`:
```php
  'debug' => true,
  'loglevel' => 0,
  'mail_smtpdebug' => true,
```

This results in the following `horde_smtp.log`: https://pastebin.com/CQfUsjS0
And the following excerpt of `nextcloud.log`: https://pastebin.com/zqj45hyd
(I've censored any references to the SMTP server, names or email addresses)

I've also tried set `'app.mail.transport' => 'php-mail',` in `config.php`, but that doesn't appear to make a difference.

Does this give any insight? If not, could you perhaps give some pointers on how to collect more info?

Mail app version: 1.4.1
Nextcloud version: 19.0.3

**Comment by agboom** [comment]
I'm having similar symptoms with sending mail. I've tried collecting some logs with the help of [General troubleshooting](https://docs.nextcloud.com/server/latest/admin_manual/issues/general_troubleshooting.html#nextcloud-logfiles) and [Logging](https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/logging_configuration.html), but I'm not getting very much. Maybe I'm missing something?

I've added these settings to `config.php`:
```php
  'debug' => true,
  'loglevel' => 0,
  'mail_smtpdebug' => true,
```

This results in the following `horde_smtp.log`: https://pastebin.com/iUxB1XZF
And the following excerpt of `nextcloud.log`: https://pastebin.com/zqj45hyd
(I've censored any references to the SMTP server, names or email addresses)

I've also tried set `'app.mail.transport' => 'php-mail',` in `config.php`, but that doesn't appear to make a difference.

Does this give any insight? If not, could you perhaps give some pointers on how to collect more info?

Mail app version: 1.4.1
Nextcloud version: 19.0.3

**Comment by agboom** [comment]
Adding to the above post: strangely enough the message sending does appear to work, but the client reports the following error and puts the sent mail into the drafts folder:

![image](https://user-images.githubusercontent.com/1269113/94342353-0ba08000-0000-11eb-942b-abc9ae2a4696.png)

**Comment by agboom** [comment]
After some more investigation, there seems to be a subtle difference between setting `'app.mail.transport' => 'php-mail'` and not setting it.

With `php-mail` the error is thrown in `/var/www/html/apps/mail/vendor/pear-pear.horde.org/Horde_Mail/Horde/Mail/Transport/Mail.php`, indicating that the result of calling the `mail()` function was false. I tried to print the error message with `error_get_last()['message']`, but that resulted in an empty string. The above excerpt of nextcloud.log shows the full stack trace (https://pastebin.com/zqj45hyd).

Moreover, with `php-mail` enabled, in my case no email is sent.

With `php-mail` disabled, the error is thrown in `/var/www/html/apps/contactsinteraction/lib/Db/CardSearchDao.php`, indicating a type error. The full stack trace is here: https://pastebin.com/8xx1c2uG

Moreover, with `php-mail` disabled, in my case the email is successfully sent, but Mail still reports the same error and puts the email in the drafts folder.

**Comment by agboom** [comment]
Hmm, my issue seems more related to this problem: https://help.nextcloud.com/t/mail-app-sending-returns-error-500-but-sending-actually-works/93253

While the symptoms are similar, I'm not sure if it has the same underlying cause as OP.

**Comment by ChristophWurst** [comment]
@agboom you're running into https://github.com/nextcloud/server/issues/21696.

**Comment by headcrushedindoor** [comment]
Same here. I get a 500 error but it delivers email in my situation.

**Comment by ant0nwax** [comment]
the user that installed the mail app to nextcloud is able to send emails,
another user is not able to send emails, also not when that second user is reaised to be admin, also that user gets 500 error on send and timeout on reading a prioritized email

**Comment by ant0nwax** [comment]
```MariaDB [(none)]> use nextcloud
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Database changed
MariaDB [nextcloud]> UPDATE oc_mail_accounts acc INNER JOIN oc_accounts ON uid=acc.user_id SET name=REGEXP_REPLACE(REGEXP_REPLACE(data, '^.*displayname":{"value":"', ''), '".*', '') WHERE name='' OR name IS NULL;
Query OK, 0 rows affected (0.001 sec)
Rows matched: 0  Changed: 0  Warnings: 0
```

**Comment by ant0nwax** [comment]
even the database records look totally okay
maybe its a user issue i must admit, i will know more on next week, now it is weekend

`|  4 | user 1   | User 1 | user1@example.com | imap.yandex.ru | 993          | ssl              | user1@example.com | 79d3a9eba7a4ba9549____e1535120690ade9ea179d4c17b5bc8ffd3c2b34c51|70099391341d6751a0230a____3f923a|4a0e063a83c7bb7b85d____7a73f137ac6e8ff7e8cd0bd2d88522da178708d6637d63c4f4986acce5e4f19bf091ea0____5f3bb7c2433b4c6e5c496fb793dd07|3 | smtp.yandex.ru | 465           | ssl               | user1@example.com | 4a04bad676986____0f02eb69e2cec718fecac78a23fe0fc119476a2b62a9177|76fe8581ed026____04a8e70ca4ab85a|b6e1aee8f2b75c54b0cd1334e42fb3f1b____5f0b8912cd8b35ca6cb55eeef6cc1064fb15ad4c782d79685c9ac27491cc____037516957c1406fe05be961c274|3 | NULL      |        16______9_| plaintext   |     1 |                    0 |                    |                18 |              21 |               23 |             0 | NULL       | NULL       | NULL           | NULL       | NULL           |            NULL |                     0 |
|  5 | user 2   | User 2     | user2@example.com | imap.yandex.ru | 993          | ssl              | user2@example.com | 77e677912f700____9999f367be43192da16bfc988b84398a2ca09a890f3b8c2|cf933d6f5c1616bc990____00cb113a0|ab11f8471783a327f0ba____ab80956ef7663808fb4736721d2ea54b02489e221cc9d247fe0e____4eaa2b8684e27f4ca173133e3bb1f8cad3ade27000d178ab|3 | smtp.yandex.ru | 465           | ssl               | user2@example.com | aa7594ecff631____113602fc0e04153f094773fb45b7e6f4bba78f97c71964c|bc8b4ccfc563a5300bb28____4f96fb2|4aa245cc816c46f988c47c58301c7ef____cadc888d881918946aa944cf16505c09acaa9c973394e8c72c073bb241f2____0372f8a2d82d1a053987b9adc36f1|3 | NULL      |        16______9_ | plaintext   |     1 |                    0 |                    |                24 |              27 |               29 |             0 | NULL       | NULL       | NULL           | NULL       | NULL           |            NULL |                     0 |

`

**Comment by ant0nwax** [comment]
by just searching i found something interesting in the SQL
how to clean that mess: lines 8 to 14 shall be removed
they contain invalid data from setting up the system first

maybe that information or some other information somewhere else destorys the mail sending experience

```
MariaDB [nextcloud]> SELECT * FROM oc_mail_tags;
+----+----------------+--------------+--------------+---------+----------------+
| id | user_id        | imap_label   | display_name | color   | is_default_tag |
+----+----------------+--------------+--------------+---------+----------------+
|  1 | user1          | $label1      | Important    | #FF7A66 |              1 |
|  2 | user1          | $label2      | Work         | #31CC7C |              1 |
|  3 | user1          | $label3      | Personal     | #A85BF7 |              1 |
|  4 | user1          | $label4      | To Do        | #317CCC |              1 |
|  5 | user1          | $label5      | Later        | #B4A443 |              1 |
|  6 | user1          | encrypted    | Encrypted    | #fff    |              0 |
|  7 | user1          | system_hamon | System Hamon | #fff    |              0 |
|  8 | User 2         | $label1      | Important    | #FF7A66 |              1 |
|  9 | User 2         | $label2      | Work         | #31CC7C |              1 |
| 10 | User 2         | $label3      | Personal     | #A85BF7 |              1 |
| 11 | User 2         | $label4      | To Do        | #317CCC |              1 |
| 12 | User 2         | $label5      | Later        | #B4A443 |              1 |
| 13 | User 2         | encrypted    | Encrypted    | #fff    |              0 |
| 14 | User 2         | system_hamon | System Hamon | #fff    |              0 |
| 15 | user2          | $label1      | Important    | #FF7A66 |              1 |
| 16 | user2          | $label2      | Work         | #31CC7C |              1 |
| 17 | user2          | $label3      | Personal     | #A85BF7 |              1 |
| 18 | user2          | $label4      | To Do        | #317CCC |              1 |
| 19 | user2          | $label5      | Later        | #B4A443 |              1 |
| 20 | user2          | encrypted    | Encrypted    | #fff    |              0 |
| 21 | user2          | system_hamon | System Hamon | #fff    |              0 |
+----+----------------+--------------+--------------+---------+----------------+
21 rows in set (0.001 sec)
```

**Comment by ant0nwax** [comment]
the user here was not capable of understanding that we need an app password in 2022 for using smtp, a user account password is not enough anymore at smtp.yandex.ru

**Comment by sazanof** [comment]
I had a similar one. In part, this happens because either the wrong recipient, or the group contains an recipient that cannot be found. 

I have a group in AD, **I disabled one member**, _**but did not throw him out of this group**_. I tried to send message to this group and saw in the Postfix logs that the _recipient is unknown_. at this moment, the 500 error in NC came out.  

Maybe it will be useful to someone
